c++/src/H5DataType.cpp

```diff
@@ -250,7 +250,7 @@ DataType::p_decode() const
     }
 
     // Call C function to decode the binary object description
-    hid_t encoded_dtype_id = H5Tdecode(encoded_buf);
+    hid_t encoded_dtype_id = H5Tdecode2(encoded_buf, buf_size);
 
```

Comment:
buf_size was initialized to 0 at creation.  Should there be a call to determine the size first?